R/Set_up_subfunctions.R

```diff
@@ -0,0 +1,338 @@
+#' Calculate proportions of pigs in each age class
+#' @description
+#' Using age stable values from life table, calculate proprtion of pigs in each age class for each state 
+#' 
+#' @param PPS pig population size
+#' @param pig_demography pig life-table
+#' @param na_pig number age classes for pigs (processed)
+#' @param IPL0_total total number of infected (low burden) pigs
+#' @param IPH0_total total number of infected (high burden) pigs
+#'
+#' @return proportions of pigs in each age class and state
+#' @export
+Pig_age_class_proportions_func <- function(PPS, pig_demography, na_pig, IPL0_total, IPH0_total){
+  
+  if (na_pig == 1){ 
+    stop("error: do not calculate prooprtions across age-classes for 1 age class (non-age strcutured model)")
+  }
+  
+  
+  PP0_total <- 0
+  RP0_total <- 0
+  VP0_total <- 0
+
+# calculate proportion/ number of pigs in each age class
+den_pig <- c()
+
+for (i in 1:na_pig) {
+  den_pig[i] <- PPS * pig_demography$age_stable[i]
+}
+
+den_pig_fraction <- den_pig / PPS
+
+# calculate numbers in each age class for each state (SP, PP, IPL, IPH, RP, VP)
+SP_eq <- c()
+for (i in 1:na_pig) {
+  SP_eq[i] <-
+    den_pig_fraction[i] * ((PPS) - (IPL0_total + IPH0_total + PP0_total + RP0_total + VP0_total))
+}
+
+PP_eq <- c()
+for (i in 1:na_pig) {
+  PP_eq[i] <- den_pig_fraction[i] * PP0_total
+}
+
+IPL_eq <- c()
+for (i in 1:na_pig) {
+  IPL_eq[i] <- den_pig_fraction[i] * IPL0_total
+}
+
+IPH_eq <- c()
+for (i in 1:na_pig) {
+  IPH_eq[i] <- den_pig_fraction[i] * IPH0_total
+}
+
+RP_eq <- c()
+for (i in 1:na_pig) {
+  RP_eq[i] <- den_pig_fraction[i] * RP0_total
+}
+
+VP_eq <- c()
+for (i in 1:na_pig) {
+  VP_eq[i] <- den_pig_fraction[i] * VP0_total
+}
+
+return(list(SP_eq, PP_eq, IPL_eq, IPH_eq, RP_eq, VP_eq))
+
+}
+
+
+#' @title
+#' Calculate key age parameters for human population
+#' @description
+#' Calculate age rate and age widths for human compartments depending on number of age groups 
+#'
+#' @param number_age_classes the number of age classes (humans)
+#'
+#' @return age rate and age width
+#' @export
+age_parameters_human_func <- function(number_age_classes){
+
+if (number_age_classes > 1) {
+  
+  # vector of specified no. of months in each age compartment 
+  if (number_age_classes == 7) {
+    age_width_human <- c()
+    age_width_human[1] <- 5 * 12 # 0 - 4.99 yrs (in months)
+    age_width_human[2] <- 5 * 12 # 5 - 9.99 yrs
+    age_width_human[3] <- 5 * 12 # 10- 14.99 yrs
+    age_width_human[4] <- 15 * 12 # 15 - 29.99 yrs
+    age_width_human[5] <- 20 * 12 # 30 - 49.99 yrs
+    age_width_human[6] <- 20 * 12 # 50 - 69.99 yrs
+    age_width_human[7] <- 20 * 12 # 70 - 89.99 yrs
+  }
+  
+  # create vector of length n age classes for ODIN (& create dimensions for other variables)
+  na_human <- as.integer(length(age_width_human)) 
+  
+  # calculate age rate (function of age width)
+  age_rate_human <- c()
+  
+  for (i in na_human) {
+    age_rate_human[1:(na_human - 1)] <- 1 / age_width_human[1:i - 1]
```

Comment:
`age_width_human` isn't defined in the function arguments. This function therefore only works if `number_age_classes = 7`. Either make it flexible to the number of age classes (as the function argument would suggest) or re-write so that it is clear only 7 age claases are supported (and remove unecessary logic).

Reply:
Have removed the `age_parameters_human_func` entirely as not really needed given model based on 7 age classes. This code has now been moved to `set_up` without the logic statements